webtau-graphql/src/main/java/org/testingisdocumenting/webtau/graphql/GraphQLCoverage.java

```diff
@@ -38,16 +38,34 @@ public void recordQuery(HttpValidationResult validationResult) {
         }
 
         Set<GraphQLQuery> graphQLQueries = schema.findQueries(validationResult.getRequestBody());
-        graphQLQueries.forEach(query -> coveredQueries.add(query, validationResult.getId(), validationResult.getElapsedTime()));
+        graphQLQueries.forEach(query -> coveredQueries.
+            add(query, validationResult.getId(), validationResult.getElapsedTime(), isErrorResult(validationResult)));
     }
 
-    Stream<GraphQLQuery> nonCoveredQueries() {
-        return schema.getSchemaDeclaredQueries().filter(o -> !coveredQueries.contains(o));
+    private boolean isErrorResult(HttpValidationResult validationResult) {
+        return JsonUtils.convertToTree(JsonUtils.deserialize(validationResult.getResponse().getTextContent()))
+            .has("errors");
     }
 
+    Stream<GraphQLQuery> nonCoveredQueries() {
+          return schema.getSchemaDeclaredQueries().filter(o -> !coveredQueries.contains(o));
+      }
+
     Stream<GraphQLQuery> coveredQueries() {
         return coveredQueries.coveredQueries();
     }
+    Stream<GraphQLQuery> coveredSuccessBranches() {
+        return coveredQueries.coveredSuccessBranches();
+    }
+    Stream<GraphQLQuery> nonCoveredSuccessBranches() {
+        return schema.getSchemaDeclaredQueries().filter(o -> coveredQueries.coveredSuccessBranches().noneMatch(graphQLQuery -> graphQLQuery.equals(o)));
```

Comment:
I suggest sticking `coveredQueries.coveredSuccessBranches()` into a local var inside this function (not in the lambda) otherwise you're doing the computation/filtration from `coveredSuccessBranches()` repeatedly.